src/views/Labels/PdfGenerator.tsx

```diff
@@ -1,94 +1,76 @@
-/* eslint-disable no-return-assign */
-/* eslint-disable react/button-has-type */
-/* eslint-disable react/no-string-refs */
-import React, { Component } from "react"
-import { Link } from "react-router-dom"
-import { PDFExport } from "@progress/kendo-react-pdf"
-import QRUpper from "../../public/Boxtribute_QR_Label_Upper.jpg"
-import QRLower from "../../public/Boxtribute_QR_Label_Lower.jpg"
 
-// eslint-disable-next-line no-unused-vars
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { PDFExport } from '@progress/kendo-react-pdf';
+import Label from './Label';
+
 class PdfGenerator extends Component<PdfGeneratorProps> {
   page
   imageUpper
   imageLower
   canvLoaded: boolean
+  pdfExportComponent!: PDFExport | null;
 
   constructor(props: PdfGeneratorProps) {
     super(props)
     this.canvLoaded = false
   }
 
   exportPDF = () => {
-    this.page.save()
-  }
+    if (this.pdfExportComponent !== null) this.pdfExportComponent.save();
+  };
 
   render() {
+    const labels : JSX.Element[] = []
+    const num = parseInt(window.location.pathname.substring(15), 10) // this.props.match.params
```

Comment:
What's the reason for reading the parameter using `window` and not with react-router? I see you tried it with `this.props.match.params`, right? I think if you use/wrap `PdfGenerator` with the `withRouter`, the variable you wanted to use will be set (https://stackoverflow.com/questions/53539314/what-is-withrouter-for-in-react-router-dom)